mountpoint-s3/src/main.rs

```diff
@@ -103,51 +99,48 @@ mod logging {
 
         RustLogAdapter::try_init().context("failed to initialize CRT logger")?;
 
-        let filename = OffsetDateTime::now_utc()
-            .format(LOG_FILE_NAME_FORMAT)
-            .context("couldn't format log file name")?;
-
-        // log directories and files should not be accessible by other users
-        let mut file_options = OpenOptions::new();
-        file_options.mode(0o640).write(true).create(true);
+        let file_layer = if let Some(path) = log_directory {
+            const LOG_FILE_NAME_FORMAT: &[FormatItem<'static>] =
+                macros::format_description!("mountpoint_s3_[year][month][day][hour][minute][second].log");
```

Comment:
Let's make this more readable while we're here
```suggestion
                macros::format_description!("mountpoint-s3-[year]-[month]-[day]-[hour]-[minute]-[second].log");
```

Reply:
How do you feel about something closer (but sadly not meeting) ISO8601?

```suggestion
                macros::format_description!("mountpoint-s3-[year]-[month]-[day]T[hour]-[minute]-[second]Z.log");
```

Example:

```
mountpoint-s3-2023-07-14T09-26-03Z.log
```

It's not quite meeting the spec, since the dashes between time should be colon. If someone really wants ISO8601, they can drop the dashes from the whole name when processing which will meet spec.

I'm also throwing in `Z` to indicate the timezone, since this timestamp is UTC.